// @aws-sdk/client-iot-data-plane/dist-es/commands/ListRetainedMessagesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRetainedMessagesRequest, ListRetainedMessagesResponse } from "../models/models_0";
import { deserializeAws_restJson1ListRetainedMessagesCommand, serializeAws_restJson1ListRetainedMessagesCommand, } from "../protocols/Aws_restJson1";
var ListRetainedMessagesCommand = (function (_super) {
    __extends(ListRetainedMessagesCommand, _super);
    function ListRetainedMessagesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRetainedMessagesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTDataPlaneClient";
        var commandName = "ListRetainedMessagesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRetainedMessagesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRetainedMessagesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRetainedMessagesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListRetainedMessagesCommand(input, context);
    };
    ListRetainedMessagesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListRetainedMessagesCommand(output, context);
    };
    return ListRetainedMessagesCommand;
}($Command));
export { ListRetainedMessagesCommand };
